make-rules/common.mk

```diff
@@ -0,0 +1,198 @@
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2015, 2016, Oracle and/or its affiliates. All rights reserved.
+#
+
+#
+# This file sets up the standard, default options and base requirements for
+# userland components.
+#
+
+include $(WS_MAKE_RULES)/prep.mk
+
+# Override this to limit builds and publication to a single architecture.
+BUILD_ARCH ?= $(MACH)
+ifneq ($(strip $(BUILD_ARCH)),$(MACH))
+PUBLISH_STAMP= 
+
+target-na:
+	@echo "Not available for $(MACH)"
+
+BUILD_TARGET ?= target-na
+INSTALL_TARGET ?= target-na
+TEST_TARGET ?= target-na
+SYSTEM_TEST_TARGET ?= target-na
+endif
+
+# Assume components use a configure script-style build by default.
+BUILD_STYLE ?= configure
+# If not an archive build (i.e. extract and copy) or pkg build (publish only),
+# include relevant makefile.
+ifneq ($(strip $(BUILD_STYLE)),archive)
+ifneq ($(strip $(BUILD_STYLE)),pkg)
+include $(WS_MAKE_RULES)/$(strip $(BUILD_STYLE)).mk
+endif
+endif
+
+ifeq ($(strip $(BUILD_STYLE)),configure)
+# Assume these items should always be set in the configure environment.  strip
+# is used to work around potential problems with extra blank characters
+# inserted into strings during nested configure invocations within builds for
+# components such as gdb.
+ifneq ($(strip $(CFLAGS)),)
+# Only if provided.
+CONFIGURE_ENV += CFLAGS="$(strip $(CFLAGS))"
+endif
+ifneq ($(strip $(CXXFLAGS)),)
+# Only if provided.
+CONFIGURE_ENV += CXXFLAGS="$(strip $(CXXFLAGS))"
+endif
+# Always process LDFLAGS.
+CONFIGURE_ENV += LDFLAGS="$(strip $(LDFLAGS))"
+endif
+
+ifeq ($(strip $(BUILD_STYLE)),justmake)
+# Assume these items should always be set in the build environment.
+COMPONENT_BUILD_ENV += CC="$(CC)"
+COMPONENT_BUILD_ENV += CXX="$(CXX)"
+ifneq ($(strip $(CFLAGS)),)
+# Only if provided.
+COMPONENT_BUILD_ENV += CFLAGS="$(strip $(CFLAGS))"
+endif
+ifneq ($(strip $(CXXFLAGS)),)
+# Only if provided.
+COMPONENT_BUILD_ENV += CXXFLAGS="$(strip $(CXXFLAGS))"
+endif
+# Always process LDFLAGS.
+COMPONENT_BUILD_ENV += LDFLAGS="$(strip $(LDFLAGS))"
+endif
+
+# Assume components built with ant do not need ASLR.
+ifeq ($(strip $(BUILD_STYLE)),ant)
+ASLR_MODE= $(ASLR_NOT_APPLICABLE)
+else
+# Enable ASLR by default unless target build is NO_ARCH.
+ifneq ($(strip $(BUILD_BITS)),NO_ARCH)
```

Comment:
We can comment out this... Perhaps, we should patch ld to respect environment variables as Oracle one does. But we likely don't want ASLR to be suddenly activated.

Reply:
Yes I was not sure how this should be handled. I am using the file for fonts only but of course this should be a discussion.